src/sys/socket/sockopt.rs

```diff
@@ -478,6 +485,67 @@ unsafe impl<'a> Set<'a, usize> for SetUsize {
     }
 }
 
+/// Size of empty CString buffer
+/// There is no default size in kernel documentation
+const SIZE_CSTRING_MAX : usize = 64;
```

Comment:
I believe the symbol you're looking for is `TCP_CA_NAME_MAX`.  You should add it to libc.

Reply:
`TCP_CA_NAME_MAX` is only for TCP_CONGESTION parameter. I wanted to make a generic string type for setsockopt / getsockopt.

Reply:
Oh, I see now.  Do you think the `GetCString` and `SetCString` structs should be generic on the string's size?

Reply:
To make size generic, we have to add an `Option<usize>` parameter to the `blank` method like: https://github.com/nix-rust/nix/pull/866/files#diff-16839e6bfe96cb0a5f52aef846eefe8eR296
Or do you have another idea?

Reply:
Can't you just do something like this?
```rust
struct GetString<T> {
    len: usize,
    val: T
}
impl<T> GetString<T> {
    unsafe fn blank() -> Self {
        GetStruct {
            len: mem::size_of::<T>() as socklen_t,
            val: mem::zeroed(),
         }
    }
}
```
Then instantiate it like `GetStruct<[u8; 16]>`?

Reply:
Done